### PR TITLE
Fix/cloud catalog api

### PR DESCRIPTION
**Pull request by ganeshkumar1989**
The calls to `/order/catalog/formatted/cloud` have been replaced with calls to the new catalog API - `/order/catalog/public/cloud`

MANAGER-3356

### CODE REVIEW

File: packages/manager/modules/cloud-universe-components/src/currency/service.js [hunk 1]
```diff
@@ -28,4 +31,12 @@ export default class CucCurrencyService {
   getCurrentCurrency() {
     return this.currency;
   }
+
+  UcentsToCurrency(value, intervalParam) {
```
Comment (marie-j) [inline packages/manager/modules/cloud-universe-components/src/currency/service.js]:
```suggestion
  UcentsToCurrency(value, intervalParam = 1) {
```
instead ? 
Moreover we should probably don't have to do that on our side or it should be

File: packages/manager/modules/pci/src/projects/project/vouchers/credit/credit-agora.controller.js [hunk 2]
```diff
@@ -30,29 +32,30 @@ export default class CloudProjectBillingVouchersAddcreditAgoraCtrl {
   $onInit() {
     this.amount = 10;
     this.loading = true;
-    return this.$http.get('/order/catalog/formatted/cloud', {
+    return this.$http.get('/order/catalog/public/cloud', {
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/vouchers/credit/credit-agora.controller.js]: Could we replace this by an `ovh-api-services` call ?

File: packages/manager/modules/cloud-universe-components/src/currency/service.js (hunk 1 above)
Comment (ganeshkumar1989) [reply packages/manager/modules/cloud-universe-components/src/currency/service.js]: That's a good idea. Thanks. This is done

File: packages/manager/modules/pci/src/projects/project/vouchers/credit/credit-agora.controller.js (hunk 2 above)
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/projects/project/vouchers/credit/credit-agora.controller.js]: This is fixed. Thanks :)

**Comment by JeremyDec** [comment]
@ganeshkumar1989 No ticket reference ?

File: packages/manager/modules/cloud-universe-components/src/currency/service.js [hunk 3]
```diff
@@ -28,4 +29,8 @@ export default class CucCurrencyService {
   getCurrentCurrency() {
     return this.currency;
   }
+
+  UcentsToCurrency(value, interval = 1) {
```
Comment (JeremyDec) [inline packages/manager/modules/cloud-universe-components/src/currency/service.js]: For clarity maybe ?
```suggestion
  convertUcentsToCurrency(value, interval = 1) {
```

File: packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js [hunk 4]
```diff
@@ -19,14 +22,9 @@ angular.module('managerApp').controller('CloudProjectBillingVouchersAddcreditAgo
...
-    return this.$http.get('/order/catalog/formatted/cloud', {
-      serviceType: 'apiv6',
-      params: {
-        ovhSubsidiary: 'US',
-      },
-    }).then((result) => {
-      this.price = get(
-        head(
+    return this.OvhApiOrderCatalogPublic.v6()
+      .get({ productName: 'cloud', ovhSubsidiary: 'US' }).then((result) => {
```
Comment (JeremyDec) [inline packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js]: Not yours, but we just want the catalog available for US ?

**Comment by ganeshkumar1989** [comment]
> @ganeshkumar1989 No ticket reference ?

Sorry... I just added it

File: packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js (hunk 4 above)
Comment (ganeshkumar1989) [reply packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js]: This particular controller gets loaded only for US. For the other regions, `cloud-project-billing-vouchers-add-credit.controller.js` gets loaded

File: packages/manager/modules/cloud-universe-components/src/currency/service.js (hunk 3 above)
Comment (ganeshkumar1989) [reply packages/manager/modules/cloud-universe-components/src/currency/service.js]: This is done. Thanks!

File: packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js (hunk 4 above)
Comment (marie-j) [reply packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js]: Even though this is only accessile for US, maybe we could use the user's subsidiary instead that way we can just change the component's restriction if needed

Comment (ganeshkumar1989) [reply packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js]: This is done. Thanks :)

File: packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js [hunk 5]
```diff
@@ -19,36 +23,36 @@ angular.module('managerApp').controller('CloudProjectBillingVouchersAddcreditAgo
...
+                head(
+                  filter(
+                    get(result, 'plans'),
+                    p => p.planCode === 'credit' && p.pricingType === 'purchase',
+                  ),
                 ),
+                'pricings',
               ),
-              'details.pricings.default',
+              p => p.capacities.indexOf('installation') >= 0,
```
Comment (FredericEspiau) [inline packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js]:
```suggestion
              p => p.capacities.includes('installation'),
```
maybe ? :)

File: packages/manager/modules/pci/src/projects/project/vouchers/credit/credit-agora.controller.js [hunk 6]
```diff
@@ -30,35 +36,35 @@ export default class CloudProjectBillingVouchersAddcreditAgoraCtrl {
...
+                head(
+                  filter(
+                    get(result, 'plans'),
+                    p => p.planCode === 'credit' && p.pricingType === 'purchase',
+                  ),
                 ),
+                'pricings',
               ),
-              'details.pricings.default',
+              p => p.capacities.indexOf('installation') >= 0,
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/vouchers/credit/credit-agora.controller.js]:
```suggestion
              p => p.capacities.includes('installation'),
```

File: packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js (hunk 5 above)
Comment (ganeshkumar1989) [reply packages/manager/apps/cloud/client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit-agora.controller.js]: This is done. Thanks :)

File: packages/manager/modules/pci/src/projects/project/vouchers/credit/credit-agora.controller.js (hunk 6 above)
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/projects/project/vouchers/credit/credit-agora.controller.js]: This is done. Thanks :)

**Comment by ganeshkumar1989** [comment]
Unblocked since the API is now in Beta

**Pull request merged**